File: programmers/Greedy/uniform.py
# n : 전체 학생 수, lost : 체육복을 도난 당한 학생들의 번호, reverse : 여벌의 체육복을 가져온 학생 번호
def solution(n, lost, reserve):
    answer = 0
    uniform_list = list([i for i in range(1,n+1)])

    for j in range(n):
        for t in range(n):
            if n == lost[j]:
                break
            else:
                uniform_list.append(t)
                break
        
        for j in len(reserve):
            if t == reserve[j]:
                break
            else:
                pass

    print(uniform_list)

    for i in len(reserve):
        for j in len(lost):
            if (reserve[len(reserve)-i-1] - lost[j] == 1):
                uniform_list.append(lost[j])

            elif reserve[i] - lost[j] == -1:
                pass
            else:
                break
                
                

        
    return answer


solution(5, [2, 4], [1, 3, 5])   # return = 5

File: programmers/Greedy/uniform.js
// function solution(n, lost, reserve) {
//   let _lost = lost.sort((a, b) => a - b);
//   let _reserve = reserve.sort((a, b) => a - b);
//   let answer = n;

//   let hasShirt = Array.from({ length: n }, () => 0);

//   for (let i = 0; i < n; i++) {
//     if (!hasShirt[i]) {
//         if (_lost[0] === i+1) {

//         }

//     }
//   }

//   return answer;
// }

function solution(n, lost, reserve) {
  let _lost = lost.sort((a, b) => a - b).filter((e) => !reserve.includes(e));
  let _reserve = reserve.sort((a, b) => a - b).filter((e) => !lost.includes(e));

  let answer = n - _lost.length;

  for (let i = 0; i < _reserve.length; i++) {
    if (!_lost.length) break;
    if (_lost.includes(_reserve[i])) {
      _lost.splice(_lost.indexOf(_reserve[i]), 1);
      ++answer;
    } else {
      if (_lost.includes(_reserve[i] - 1)) {
        _lost.splice(_lost.indexOf(_reserve[i] - 1), 1);
        ++answer;
      } else if (_lost.includes(_reserve[i] + 1)) {
        _lost.splice(_lost.indexOf(_reserve[i] + 1), 1);
        ++answer;
      }
    }
  }

  return answer;
}

let n = 5;
let lost = [4, 3];
let reserve = [3, 2];

solution(n, lost, reserve);
